File: graphql-dgs-codegen-core/src/integTest/java/com/netflix/graphql/dgs/codegen/java/testcases/unions/unionTypesWithoutInterfaceCanDeserialize/expected/types/Droid.java
package com.netflix.graphql.dgs.codegen.java.testcases.unions.unionTypesWithoutInterfaceCanDeserialize.expected.types;

import com.fasterxml.jackson.annotation.JsonTypeInfo;
import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import java.util.Objects;

@JsonTypeInfo(
    use = JsonTypeInfo.Id.NONE
)
public class Droid implements SearchResult {
  private String id;

  private String name;

  private String primaryFunction;

  public Droid() {
  }

  public Droid(String id, String name, String primaryFunction) {
    this.id = id;
    this.name = name;
    this.primaryFunction = primaryFunction;
  }

  public String getId() {
    return id;
  }

  public void setId(String id) {
    this.id = id;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getPrimaryFunction() {
    return primaryFunction;
  }

  public void setPrimaryFunction(String primaryFunction) {
    this.primaryFunction = primaryFunction;
  }

  @Override
  public String toString() {
    return "Droid{id='" + id + "', name='" + name + "', primaryFunction='" + primaryFunction + "'}";
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    Droid that = (Droid) o;
    return Objects.equals(id, that.id) &&
        Objects.equals(name, that.name) &&
        Objects.equals(primaryFunction, that.primaryFunction);
  }

  @Override
  public int hashCode() {
    return Objects.hash(id, name, primaryFunction);
  }

  public static Builder newBuilder() {
    return new Builder();
  }

  public static class Builder {
    private String id;

    private String name;

    private String primaryFunction;

    public Droid build() {
      Droid result = new Droid();
      result.id = this.id;
      result.name = this.name;
      result.primaryFunction = this.primaryFunction;
      return result;
    }

    public Builder id(String id) {
      this.id = id;
      return this;
    }

    public Builder name(String name) {
      this.name = name;
      return this;
    }

    public Builder primaryFunction(String primaryFunction) {
      this.primaryFunction = primaryFunction;
      return this;
    }
  }
}

File: graphql-dgs-codegen-core/src/integTest/kotlin/com/netflix/graphql/dgs/codegen/cases/unionTypesWithoutInterfaceCanDeserialize/expected/types/Droid.kt
package com.netflix.graphql.dgs.codegen.cases.unionTypesWithoutInterfaceCanDeserialize.expected.types

import com.fasterxml.jackson.`annotation`.JsonIgnoreProperties
import com.fasterxml.jackson.`annotation`.JsonProperty
import com.fasterxml.jackson.`annotation`.JsonTypeInfo
import com.fasterxml.jackson.databind.`annotation`.JsonDeserialize
import com.fasterxml.jackson.databind.`annotation`.JsonPOJOBuilder
import com.netflix.graphql.dgs.codegen.cases.unionTypesWithoutInterfaceCanDeserialize.expected.Generated
import java.lang.IllegalStateException
import kotlin.Any
import kotlin.Boolean
import kotlin.Int
import kotlin.String
import kotlin.collections.List
import kotlin.jvm.JvmName

@Generated
@JsonTypeInfo(use = JsonTypeInfo.Id.NONE)
@JsonDeserialize(builder = Droid.Builder::class)
public class Droid(
  id: () -> String = idDefault,
  name: () -> String = nameDefault,
  primaryFunction: () -> String? = primaryFunctionDefault,
) : SearchResult {
  private val __id: () -> String = id

  private val __name: () -> String = name

  private val __primaryFunction: () -> String? = primaryFunction

  @get:JvmName("getId")
  public val id: String
    get() = __id.invoke()

  @get:JvmName("getName")
  public val name: String
    get() = __name.invoke()

  @get:JvmName("getPrimaryFunction")
  public val primaryFunction: String?
    get() = __primaryFunction.invoke()

  private fun `__$fieldValues`(): List<Any?> = listOf(
      if (__id === idDefault) idDefault else id,
      if (__name === nameDefault) nameDefault else name,
      if (__primaryFunction === primaryFunctionDefault) primaryFunctionDefault else primaryFunction,
  )

  override fun equals(other: Any?): Boolean = this === other || (other is Droid &&
      `__$fieldValues`() == other.`__$fieldValues`())

  override fun hashCode(): Int = `__$fieldValues`().hashCode()

  private fun `__$fieldStrings`(): List<String> = listOfNotNull(
      if (__id === idDefault) null else "id=" + id,
      if (__name === nameDefault) null else "name=" + name,
      if (__primaryFunction === primaryFunctionDefault) null else "primaryFunction=" +
      primaryFunction,
  )

  override fun toString(): String = `__$fieldStrings`().joinToString(prefix = "Droid(", postfix =
      ")")

  @Generated
  public companion object {
    private val idDefault: () -> String = 
        { throw IllegalStateException("Field `id` was not requested") }

    private val nameDefault: () -> String = 
        { throw IllegalStateException("Field `name` was not requested") }

    private val primaryFunctionDefault: () -> String? = 
        { throw IllegalStateException("Field `primaryFunction` was not requested") }
  }

  @Generated
  @JsonPOJOBuilder
  @JsonIgnoreProperties("__typename")
  public class Builder {
    private var id: () -> String = idDefault

    private var name: () -> String = nameDefault

    private var primaryFunction: () -> String? = primaryFunctionDefault

    @JsonProperty("id")
    public fun withId(id: String): Builder = this.apply {
      this.id = { id }
    }

    @JsonProperty("name")
    public fun withName(name: String): Builder = this.apply {
      this.name = { name }
    }

    @JsonProperty("primaryFunction")
    public fun withPrimaryFunction(primaryFunction: String?): Builder = this.apply {
      this.primaryFunction = { primaryFunction }
    }

    public fun build(): Droid = Droid(
      id = id,
      name = name,
      primaryFunction = primaryFunction,
    )
  }
}
